MvvmCross-Forms/MvvmCross.Forms.Droid/Platform/MvxFormsAndroidSetup.cs

```diff
@@ -51,13 +53,16 @@ public MvxFormsApplication FormsApplication
                     var activity = Mvx.Resolve<IMvxAndroidCurrentTopActivity>().Activity ?? ApplicationContext;
                     Xamarin.Forms.Forms.Init(activity, null);
                 }
+
                 if (_formsApplication == null)
-                    _formsApplication = CreateFormsApplication();
+                {
+                    _formsApplication = _formsApplication ?? CreateFormsApplication();
+                }
                 return _formsApplication;
             }
         }
 
-        protected virtual MvxFormsApplication CreateFormsApplication() => new MvxFormsApplication();
+        protected virtual TForms CreateFormsApplication() => new TForms();
```

Comment:
Maybe we should make this abstract if it gives bugs?

Reply:
As per my other comment, I think we should revert this change. In the generic setup class the CreateFormsApplication can return a new instance of the generic TForms class